core/src/display_object/movie_clip.rs

```diff
@@ -2843,20 +2864,55 @@ impl<'gc> TDisplayObject<'gc> for MovieClip<'gc> {
             .unwrap_or(Avm2Value::Null)
     }
 
+    /// Returns object2()- however, if the MC was AVM1, creates and returns an AVM1Movie.
+    fn object2_possibly_avm1(&self, context: &mut UpdateContext<'_, 'gc>) -> Avm2Value<'gc> {
```

Comment:
does this need to be lazy? Something in line with the "normal" construction, like:
```diff
 fn construct_frame(&self, context: &mut UpdateContext<'_, 'gc>) {
    if self.movie().is_action_script_3() && self.frames_loaded() >= 1 {
        self.allocate_as_avm2_object(context, (*self).into());
+   } else if !self.movie().is_action_script_3() && context.stage.is_action_stage_3() && self.frames_loaded() >= 1 {
+       // self.object2 = new AVM1Movie();              
```

Reply:
The problem is that a _ton_ of places check if `object2` is null to check if the SWF is AVM1. However, #12975 got merged so i need to rewrite this part anyway.